test/CMakeLists.txt

```diff
@@ -30,14 +30,15 @@ add_definitions(-DGTEST_DONT_DEFINE_ASSERT_GT=1)
 
 execute_process(COMMAND cmake -E remove_directory ${CMAKE_BINARY_DIR}/test_results)
 execute_process(COMMAND cmake -E make_directory ${CMAKE_BINARY_DIR}/test_results)
-include_directories(${GTEST_INCLUDE_DIRS})
+include_directories(SYSTEM ${GTEST_INCLUDE_DIRS})
```

Comment:
Why is this change necessary?

Reply:
Including directories as SYSTEM directories suppresses compiler warnings in includes from dependencies. More info [here](https://cmake.org/cmake/help/v3.0/command/include_directories.html) and [here](https://stackoverflow.com/a/26297802).